core/app/models/spree/address.rb

```diff
@@ -148,14 +148,14 @@ def active_merchant_hash
     end
 
     # @todo Remove this from the public API if possible.
-    # @return [true] whether or not the address requires a phone number to be
-    #   valid
+    # @return [false] whether or not the address requires a phone number to be
+    #   present
     def require_phone?
-      true
+      false
```

Comment:
What do we do for stores that relied on this being `true` by default? If I require the phone number in my store, I have not overridden `require_phone?`. When I update Solidus, the phone will become optional without me having touched anything, which is not ideal.

I think we should formally deprecate this. We can introduce a new preference called `require_phone_on_address` (or just `require_phone`?) which this method relies on.

The default value for the preference would be `nil` and, when it's `nil`, we cast it to `true` and print a deprecation warning notifying the user that they need to set an explicit value, because the new behavior in 3.0 will be for phone numbers to be optional.

In 3.0, we just switch the default value for the preference to `false`, and the phone number will become optional.

@kennyadsl is our deprecation expert, so he should also chime in probably. 😜

Reply:
Sorry for the late response here - the new version of the PR addresses this with a first pass at deprecation. How does it look now?